stream-projects/do-something/components/action/action.tsx

```diff
@@ -0,0 +1,83 @@
+import { Avatar, React, useStyleSheet } from "../../deps.ts";
+import styleSheet from "./action.scss.js";
+import { Action } from "../../shared/types/action.ts";
+import { timeAgo } from "../../shared/util/time-ago.ts";
+
+type StringObj = Record<string, string>;
+type Styles<T extends string> = Record<T, StringObj>;
+type State = "recently-redeemed" | "last-redeemed" | "normal";
+type Mode = "normal" | "recent";
+
+const stateStyleMap: Styles<State> = {
+  // this state is active for 8 seconds before switching to "last-redeemed"
+  "recently-redeemed": {
+    "--background": "var(--color-primary)",
+    "--text-color-normal": "var(--color-text-bg-light)",
+    "--text-color-demphasized": "rgba(0, 0, 0, 0.8)",
+  },
+  "last-redeemed": {
+    "--background": "var(--color-bg-tertiary)",
+    "--text-color-normal": "var(--color-text-bg-dark)",
+    "--text-color-demphasized": "rgba(255, 255, 255, 0.8)",
+  },
+  "normal": {
+    "--background": "var(--color-bg-primary)",
+    "--text-color-normal": "var(--color-text-bg-dark)",
+    "--text-color-demphasized": "rgba(255, 255, 255, 0.8)",
+  },
+};
+
+const modeStyleMap: Styles<Mode> = {
+  "normal": {
+    "--font-size-normal": "16px",
+    "--font-weight-normal": "400",
+    "--font-size-demphasized": "14px",
+    "--font-weight-demphasized": "400",
+    "--padding": "8px 24px",
+    "--avatar-size": "32px",
+    "--collectible-icon-size": "24px",
+  },
+  "recent": {
+    "--font-size-normal": "18px",
+    "--font-weight-normal": "600",
+    "--font-size-demphasized": "16px",
+    "--font-weight-demphasized": "400",
+    "--padding": "20px 24px",
+    "--avatar-size": "40px",
+    "--collectible-icon-size": "32px",
+  },
+};
+
+export default function Action(
```

Comment:
Think it would be worth renaming this to something more specific since we have an `Action` model that we use pretty frequently now. Maybe `RedeemableAction`, `DoSomethingAction`, `AlertAction`, or something else?

Reply:
Yeah that's a good idea. I like `DoSomethingAction`.

Reply:
Update: https://github.com/trufflehq/truffle-packages/pull/51/commits/f3f31f23187960bf11d76b2c0436431d18d99b97